compiler/rustc_hir/src/hir.rs

```diff
@@ -344,6 +344,39 @@ impl GenericArgs<'_> {
 
         own_counts
     }
+
+    pub fn span(&self) -> Option<Span> {
+        self.args
+            .iter()
+            .filter(|arg| !arg.is_synthetic())
+            .map(|arg| arg.span())
+            .fold_first(|span1, span2| span1.to(span2))
+    }
+
+    /// Returns span encompassing arguments and their surrounding `<>` or `()`
+    pub fn span_ext(&self, sm: &SourceMap) -> Option<Span> {
+        let mut span = self.span()?;
+
+        let (o, c) = if self.parenthesized { ('(', ')') } else { ('<', '>') };
+
+        if let Ok(snippet) = sm.span_to_snippet(span) {
+            let snippet = snippet.as_bytes();
+
+            if snippet[0] != (o as u8) || snippet[snippet.len() - 1] != (c as u8) {
+                span = sm.span_extend_to_prev_char(span, o, true);
+                span = span.with_lo(span.lo() - BytePos(1));
+
+                span = sm.span_extend_to_next_char(span, c, true);
+                span = span.with_hi(span.hi() + BytePos(1));
+            }
```

Comment:
Does this work also for the cases `< T, K>` and `<T, K >`?

Reply:
Yes - I've dropped a test at `unboxed-closure-sugar-wrong-number-number-type-parameters.rs`.

`span_extend_to_prev_char()` & `span_extend_to_next_char()` return spans _up to_ given characters, but not including them, so without `+/- BytePos(1)`, we'd end up with:

```
MyType <   foo   >
        ^^^^^^^^^
```

Another solution would be to pass some flag to `span_extend_*()` or create another two functions, but - on the other hand - anyone curious can just drop `-/+ BytePos(1)` and see which tests fail.